lib/internal/streams/pipeline.js

```diff
@@ -48,7 +48,7 @@ function destroyer(stream, reading, writing, callback) {
 
     destroyImpl.destroyer(stream, err);
 
-    callback(err || new ERR_STREAM_DESTROYED('pipe'));
```

Comment:
@mafintosh @mcollina: What is the thought behind this? It seems to cause unnecessary errors at the moment?

Reply:
What types of errors is this causing? By calling the destroyed to tear down the pipeline,  we are making sure that the callback at the end is called with an error - a pipeline that is teared down before "normal" conclusion is a notable fact and an error condition. I'd be fine if this error is only created once for the pipeline and not once for every stream (same behavior, less number of Error objects)

Reply:
This test fails:

```js
+ Error [ERR_STREAM_DESTROYED]: Cannot call pipe after a stream was destroyed
+     at internal/streams/pipeline.js:51:21
+     at finish (internal/streams/pipeline.js:144:9)
+     at internal/streams/pipeline.js:202:15
+     at processTicksAndRejections (internal/process/task_queues.js:97:5) {
+   code: 'ERR_STREAM_DESTROYED'
+ }
- undefined
    at /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-stream-pipeline.js:764:12
    at /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-stream-pipeline.js:21:23
    at new Promise (<anonymous>)
    at /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-stream-pipeline.js:21:5
    at /Users/ronagy/GitHub/nxtedition/node/test/common/index.js:362:15
    at internal/util.js:392:14
    at finish (internal/streams/pipeline.js:149:7)
    at internal/streams/pipeline.js:155:7
    at internal/util.js:392:14
    at internal/streams/pipeline.js:51:5 {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: [NodeError],
  expected: undefined,
  operator: 'strictEqual'
}
```

Note that we did not notice this before because of https://github.com/nodejs/node/pull/31835#discussion_r381108454

Reply:
I will dig deeper into this and ping you.  Meanwhile, please take a look at the linked comment.